os/board/rtl8721csm/src/component/os/tizenrt/rtk_netmgr.c

```diff
@@ -82,15 +82,98 @@ static trwifi_scan_list_s *g_scan_list;
 static int g_scan_num;
 extern struct netdev *ameba_nm_dev_wlan0;
 
+struct ap_scan_list {
+	unsigned int channel;
+	char ssid[TRWIFI_SSID_LEN + 1];
+	unsigned int ssid_length;
+	int rssi;
+	trwifi_ap_auth_type_e ap_auth_type;
+	trwifi_ap_crypto_type_e ap_crypto_type;
+};
+typedef struct ap_scan_list ap_scan_list_s;
+
+static ap_scan_list_s *saved_scan_list;
+static uint32_t scan_number = 0;
+static uint32_t ap_channel;
+
+#include "freertos/wrapper.h"
+#include "osdep_service.h"
+#define SCAN_TIMER_DURATION 180000
+static _timer scan_timer;
+static _mutex scanlistbusy;

```

Comment:
it would be better to comment what shared variables does this protect to.

Reply:
Done.